File: 9000/Baekjoon_No9012.py
data_count = int(input())

for _ in range(data_count):
    data = input()
    result = 0
    for i in range(len(data)):
        if data[i] == '(':
            result += 1
        elif data[i] == ')':
            result -= 1
        if result == -1:
            print("NO")
            break
        elif i == len(data)-1:
            if result == 0:
                print("YES")
                break
            else:
                print("NO")
                break

File: 9000/Baekjoon_No9012.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;

public class b9012 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            Queue<String> open = new LinkedList<>();
//            Queue<String> close = new LinkedList<>();
            String result = "YES";
            String data = br.readLine();

            for (int j = 0; j < data.length(); j++) {
                if (data.charAt(j) == '(') {
                    open.offer("(");
                }
                else if (data.charAt(j) == ')') {
                    if (!open.isEmpty() && open.peek().equals("(")) {
                        open.poll();
                    }
                    else {
                        result = "NO";
                        break;
                    }
                }
            }
            if (!open.isEmpty())
                result = "NO";
            System.out.println(result);
        }
    }
}
